Thank you @nashif for working on this.

I am afraid there are more issues with this setting. Trying it with
```
sanitycheck -x=CONFIG_KERNEL_COHERENCE=y -v -T tests/kernel
```
results in tests failing to build:
```
kernel/init.c:161:36: error: incompatible type for argument 1 of 'arch_mem_coherent'
  161 |  __ASSERT_NO_MSG(arch_mem_coherent(_kernel));

...

kernel/include/kswap.h:111:3: error: implicit declaration of function 'arch_cohere_stacks' [-Werror=implicit-function-declaration]
  111 |   arch_cohere_stacks(old_thread, NULL, new_thread);
```

--- a/kernel/Kconfig
+++ b/kernel/Kconfig
@@ -842,7 +842,8 @@ config TRACE_SCHED_IPI
 
 config KERNEL_COHERENCE
 	bool "Place all shared data into coherent memory"
-	default y if ARCH_HAS_COHERENCE && SMP && MP_NUM_CPUS > 1
+	depends on ARCH_HAS_COHERENCE
+	default y if SMP && MP_NUM_CPUS > 1
 	select THREAD_STACK_INFO
 	help
 	  When available and selected, the kernel will build in a mode

